src/main/java/edu/byu/cs/autograder/score/Scorer.java

```diff
@@ -33,11 +34,18 @@ public class Scorer {
      * The penalty to be applied per day to a late submission.
      * This is out of 1. So putting 0.1 would be a 10% deduction per day
      */
-    private static final float PER_DAY_LATE_PENALTY = 0.1F;
+    private final float PER_DAY_LATE_PENALTY;
     private final GradingContext gradingContext;
 
     public Scorer(GradingContext gradingContext) {
         this.gradingContext = gradingContext;
+        try {
+            ConfigurationDao dao = DaoService.getConfigurationDao();
+            PER_DAY_LATE_PENALTY = dao.getConfiguration(ConfigurationDao.Configuration.PER_DAY_LATE_PENALTY, Float.class);
+        } catch (DataAccessException e) {
+            LOGGER.error("Error while getting Per Day Late Penalty for Scorer.");
+            throw new RuntimeException(e);
```

Comment:
How far up the chain could this `RuntimeException` go? If the field were missing due to a misconfigurstion, would that result in crashing the autograder?

Reply:
Also, because of how the config system was set up, in almost every cause it will return some sort of default value instead of throwing if the value wasn't set.

Reply:
That's a good point, James, I didn't see that. Dallin, you are incorrect of how far that exception would go. Might this be replaceable with a GradingException?

Reply:
Also, now that it's not a constant, might this be able to go closer to where it's actually used?

Reply:
I'm sure we could do that. However, I might be wrong in the steps but I'm correct in its effect. I ran the code locally with a change that forced it to throw (just for testing) and nothing crashed, the grader just stopped.

Reply:
Ok, so this still isn't breaking. We can rename the exception if it makes you happier, but the code is stable as is.

Reply:
[Insert personal vendetta against unchecked exceptions here]

Reply:
> [Insert personal vendetta against unchecked exceptions here]

I also am against unchecked exceptions. When I originally wrote the code, I had a bad assumption about when that line was run, so I thought a RunTimeException was appropriate. It wasn't until @frozenfrank asked about it that I specifically tested that and realized when it ran. Making it a checked exception would be great.

> In this case, it may be appropriate to introduce a new ConfigValueNotFound exception

I'm not sure this is the appropriate name or exception. If there's a DataAccessException thrown while accessing a config value, it truly is an issue with Data Access. All possible config keys exist in the database at startup (if they're new, they have a value of `default/changeme` and each class has some sort of default value. See #485) The only scenarios I can imagine this throwing is if the database gets disconnected, or someone drops the table while its running.

Reply:
@ThanGerlek was that meant as you though James and or myself had a vendetta or that you do?

Reply:
No, only that I do. Sorry, lol